src/mapper/pkg/resolvers/helpers.go

```diff
@@ -85,22 +62,16 @@ func (i *IntentsHolder) Reset() {
 	i.lock.Lock()
 	defer i.lock.Unlock()
 
-	i.store = make(intentsHolderStore)
+	i.store.Reset()
 }
 
-func (i *IntentsHolder) AddIntent(srcService model.OtterizeServiceIdentity, dstService model.OtterizeServiceIdentity) {
+func (i *IntentsHolder) AddIntent(srcService model.OtterizeServiceIdentity, dstService model.OtterizeServiceIdentity, discoveryTime time.Time) {
 	i.lock.Lock()
 	defer i.lock.Unlock()
 
-	intents, ok := i.store[srcService]
-	if !ok {
-		intents = goset.NewSet[model.OtterizeServiceIdentity]()
-		i.store[srcService] = intents
-	}
-	intent := model.OtterizeServiceIdentity{Name: dstService.Name, Namespace: dstService.Namespace}
-	if !intents.Contains(intent) {
+	storeUpdated := i.store.Update(srcService, dstService, discoveryTime)
+	if storeUpdated {
```

Comment:
storeUpdated is only used to determine the lastIntentsUpdate time, but actually, that makes lastIntentsUpdate be logic that is split between IntentsHolder and IntentsHolderStore.

This made it hard for me to understand what this does - why do we need this boolean? when is the discoveryTime earlier than what we've already seen? I think that should never happen, no?